daemon/logger/awslogs/cloudwatchlogs.go

```diff
@@ -192,8 +219,50 @@ func (l *logStream) Close() error {
 	return nil
 }
 
-// create creates a log stream for the instance of the awslogs logging driver
+// create creates log group and log stream for the instance of the awslogs logging driver
 func (l *logStream) create() error {
+	if err := l.createLogStream(); err != nil {
+		if l.logCreateGroup {
+			if awsErr, ok := err.(awserr.Error); ok && awsErr.Code() == resourceNotFoundCode {
```

Comment:
> Calling `DescribeLogGroups` will require the `logs:DescribeLogGroups` permission in the user's IAM policy

Ah, right, so _checking_ if a group exists actually requires different permissions than _using_ a group. Hm, yes, In that case we should go with the current implementation.

I was looking if we could reduce the amount of nested `if`'s, for example

```go
	err := l.createLogStream()
	if err == nil {
		return nil
	}
```

But looking at that, I'm not sure it improves readability, so :smile: